packages/datetime/src/datePicker.tsx

```diff
@@ -264,6 +283,12 @@ export class DatePicker extends AbstractPureComponent<IDatePickerProps, IDatePic
         this.updateValue(value, true);
     };
 
+    private handleTimeChange = (time: Date) => {
+        Utils.safeInvoke(this.props.timePickerProps.onChange, time);
+        const newValue = DateUtils.getDateTime(this.state.value || new Date(), time);
```

Comment:
nit: I prefer explicit boolean expressions. Makes the intent clear, and false-y values like `0` can throw off these shorthand expressions (less of an issue with timestamps that will likely never be `0` on purpose, but still).

e.g.
```tsx
const { value } = this.state.value;
const date = value != null ? value : new Date();
const newValue = DateUtils.getDateTime(date, time);
```